multinet/uploaders/csv.py

```diff
@@ -76,39 +104,33 @@ def validate_csv(
         data_errors.append(KeyFieldAlreadyExists(key=key_field))
         raise ValidationFailed(data_errors)
 
-    if key_field in fieldnames:
-        # Node Table, check for key uniqueness
-        keys = [row[key_field] for row in rows]
-        unique_keys: Set[str] = set()
-        for key in keys:
-            if key in unique_keys:
-                data_errors.append(DuplicateKey(key=key))
-            else:
-                unique_keys.add(key)
-
-    elif "_from" in fieldnames and "_to" in fieldnames:
-        # Edge Table, check that each cell has the correct format
-        valid_cell = re.compile("[^/]+/[^/]+")
-
-        for i, row in enumerate(rows):
-            fields: List[str] = []
-            if not valid_cell.match(row["_from"]):
-                fields.append("_from")
-            if not valid_cell.match(row["_to"]):
-                fields.append("_to")
-
-            if fields:
-                # i+2 -> +1 for index offset, +1 due to header row
-                data_errors.append(InvalidRow(fields=fields, row=i + 2))
-
-    else:
-        # Unsupported Table, error since we don't know what's coming in
-        data_errors.append(UnsupportedTable())
+    keys = [row[key_field] for row in rows]
```

Comment:
This should be a regular iterator, rather than a list:

```suggestion
    keys = (row[key_field] for row in rows)
```

Reply:
Good point